### PR TITLE
🆒 vanity url

### PR DESCRIPTION
**Pull request by JUtupe**

### CODE REVIEW

File: src/main/kotlin/pl/newbies/event/infrastructure/repository/Events.kt [hunk 1]
```diff
@@ -18,6 +19,8 @@ object Events : StringNanoIdTable() {
     val subtitle = varchar("subtitle", length = 100, collate = "utf8_general_ci").nullable()
     val author = reference("author", Users)
 
+    val vanityUrl = varchar("vanityUrl", length = 50, collate = "utf8_general_ci").uniqueIndex()
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/infrastructure/repository/Events.kt]: no need for uniqueIndex

File: src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt [hunk 2]
```diff
@@ -19,8 +22,16 @@ data class EventRequest(
 
     init {
         validate(this) {
-            validate(EventRequest::title).isNotBlank()
-            validate(EventRequest::tags).distinct()
+            validate(EventRequest::title)
+                .isNotBlank()
+                .hasSize(min = 10, max = 100)
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: To much for min. 
for scale: `1234567890`

(some short names u know)
ex. "BarCoders" or "świt"

File: src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt [hunk 3]
```diff
@@ -40,8 +51,14 @@ data class AddressRequest(
...
-            validate(AddressRequest::city).isNotBlank()
-            validate(AddressRequest::place).isNotBlank()
+            validate(AddressRequest::city)
+                .isNotBlank()
+                .hasSize(max = 50)
+                .oneLine()
+            validate(AddressRequest::place)
+                .isNotBlank()
+                .hasSize(max = 50)
+                .oneLine()
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: I'm not sure if one line in place is enough to store address 
we don't have full address  combo in database
only city and place

File: src/main/kotlin/pl/newbies/event/domain/service/EventService.kt [hunk 4]
```diff
@@ -149,6 +124,35 @@ class EventService {
...
+            }
+        }
+    }
+
+    private fun getEventVanityUrl(title: String): String =
+        StringUtils.stripAccents(title)
+            .filter { it.isLetterOrDigit() || it in listOf(' ') }
+            .replace(Regex("[ \\t]{2,}"), " ") // multiple spaces to one space
+            .trim()
+            .replace(' ', '-')
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/domain/service/EventService.kt]: why not to replace `/\b[ ]+\b/` with `-` right away?

File: src/test/kotlin/pl/newbies/event/EventTest.kt [hunk 5]
```diff
@@ -732,6 +818,20 @@ class EventTest : IntegrationTest() {
...
+        fun vanityUrlTestCases() = listOf(
+            VanityUrlTestCase("Jak skutecznie japko", "jak-skutecznie-japko"),
+            VanityUrlTestCase("spaceonlastchar   ", "spaceonlastchar"),
+            VanityUrlTestCase("illegallastchar   <", "illegallastchar"),
+            VanityUrlTestCase("123 - testowa N4zwa", "123-testowa-n4zwa"),
+            VanityUrlTestCase("345-testowa N4zwa", "345testowa-n4zwa"),
+            VanityUrlTestCase("testowe  wydarzenie", "testowe-wydarzenie"),
+            VanityUrlTestCase("somerandomwithoutspaces", "somerandomwithoutspaces"),
+            VanityUrlTestCase("/to/do k,o'm`entarz", "todo-komentarz"),
+            VanityUrlTestCase("ąąłłó óććęę", "aallo-occee"),
```
Comment (DiD3n) [inline src/test/kotlin/pl/newbies/event/EventTest.kt:907]: what about non polish's weird chars `Ää Ãã Āā Áá Àà` ?

**Review by DiD3n — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pl/newbies/event/infrastructure/repository/Events.kt (hunk 1 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/infrastructure/repository/Events.kt]: I forgor 💀

File: src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt (hunk 2 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: I'll set min to 3 letters then

File: src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt (hunk 3 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: But do we need more? We can omit postal code and everything else cause user can pick coordinates on map, I think that this field will be used more like "how to get there" tip. What do you suggest?

File: src/main/kotlin/pl/newbies/event/domain/service/EventService.kt (hunk 4 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/domain/service/EventService.kt]: Oh, I can pass what characters to trim inside the `trim()` function. I'll change that

File: src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt (hunk 3 above)
Comment (DiD3n) [reply src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: i think "how to get there" field would greatly profit from multi line "how to get there" instructions

Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: Done

File: src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt [hunk 6]
```diff
@@ -40,8 +51,13 @@ data class AddressRequest(
 
     init {
         validate(this) {
-            validate(AddressRequest::city).isNotBlank()
-            validate(AddressRequest::place).isNotBlank()
+            validate(AddressRequest::city)
+                .isNotBlank()
+                .hasSize(max = 50)
+                .maxLines(5)
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: i don't know any 5 line city

File: src/test/kotlin/pl/newbies/event/EventTest.kt (hunk 5 above)
Comment (JUtupe) [reply src/test/kotlin/pl/newbies/event/EventTest.kt:907]: I've added more tests, all non-standard characters should be now handled or removed

File: src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt (hunk 6 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/application/model/EventRequest.kt]: bruh, I've changed wrong field. That's what you get if you do something in a hurry :/

**Pull request merged**